src/keys/rsa-class.js

```diff
@@ -102,104 +93,67 @@ class RsaPrivateKey {
    * The public key is a protobuf encoding containing a type and the DER encoding
    * of the PKCS SubjectPublicKeyInfo.
    *
-   * @param {function(Error, id)} callback
-   * @returns {undefined}
+   * @returns {Promise<String>}
    */
-  id (callback) {
-    this.public.hash((err, hash) => {
-      if (err) {
-        return callback(err)
-      }
-      callback(null, bs58.encode(hash))
-    })
+  async id () {
+    const hash = await this.public.hash()
+    return bs58.encode(hash)
   }
 
   /**
    * Exports the key into a password protected PEM format
    *
    * @param {string} [format] - Defaults to 'pkcs-8'.
    * @param {string} password - The password to read the encrypted PEM
-   * @param {function(Error, KeyInfo)} callback
-   * @returns {undefined}
+   * @returns {KeyInfo}
    */
-  export (format, password, callback) {
-    if (typeof password === 'function') {
-      callback = password
+  async export (format, password) { // eslint-disable-line require-await
```

Comment:
Given that format is optional, would it make sense to reverse the argument order?

Reply:
Yes, but it'd be good to limit the scope of this PR to converting the API to async/await rather than general API improvements.

Reply:
I've created an issue for this: #153